modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/ConcatWorkflowOperationHandler.java

```diff
@@ -192,39 +209,42 @@ private WorkflowOperationResult concat(MediaPackage src, WorkflowOperationInstan
     if (profile == null)
       throw new WorkflowOperationException("Encoding profile '" + encodingProfile + "' was not found");
 
-    // Output resolution
-    if (outputResolution == null)
-      throw new WorkflowOperationException("Output resolution must be set!");
+    // Output resolution - if not keeping dimensions the same, it must be set
+    if (!sameCodec && outputResolution == null)
+        throw new WorkflowOperationException("Output resolution must be set!");
 
     Dimension outputDimension = null;
-    if (outputResolution.startsWith(OUTPUT_PART_PREFIX)) {
-      if (!trackSelectors.keySet().contains(
-              Integer.parseInt(outputResolution.substring(OUTPUT_PART_PREFIX.length()))))
-        throw new WorkflowOperationException("Output resolution part not set!");
-    } else {
-      try {
-        String[] outputResolutionArray = StringUtils.split(outputResolution, "x");
-        if (outputResolutionArray.length != 2) {
-          throw new WorkflowOperationException("Invalid format of output resolution!");
+    if (!sameCodec) { // Ignore resolution if same Codec - no scaling
+      if (outputResolution.startsWith(OUTPUT_PART_PREFIX)) {
+        if (!trackSelectors.keySet().contains(
+            Integer.parseInt(outputResolution.substring(OUTPUT_PART_PREFIX.length()))))
+          throw new WorkflowOperationException("Output resolution part not set!");
```

Comment:
Incorrect error message.  In this case it's not *found*.  It may or may not be set.  Ideally we check if the prefix is set earlier.

Reply:
I'm also somewhat confused what we're doing here.

If we have (potentially) mismatched inputs
if our output resoltion starts with the output partition prefix?
if the track selector keyset does not contain a parsable integer within the output resolution with the partition prefix removed?
throw an exception

What's the goal?  And are the variables messed up or am I misreading things?